### ISSUE TITLE
No irq handler for vector

### ISSUE THREAD

**Issue by jasonmccallister** [canonical/multipass#1778]
**Describe the bug**
I have been using multipass for a while and I started getting these alerts randomly (they seem to have increased in frequency in the last week). We built a wrapper CLI around multipass for local development: https://craftcms.com/blog/nitro-1-0-released

Not sure where to start troubleshooting, but I have had multiple reports from our users that they are seeing the same somewhat randomly.

**To Reproduce**
1. `multipass exec nitro-dev -- journalctl -u nitrod -f`

**Expected behavior**
The content of journalctl are output, but then I start receiving terminal bells with: 

```
Message from syslogd@nitro-dev at Oct  2 12:13:49 ...
 kernel:[   70.010266] do_IRQ: 3.37 No irq handler for vector
```

**Additional info**
- OS: macOS 10.15.6
- multipass  1.4.0+mac
- multipassd 1.4.0+mac

**Comment by townsend2010** [comment]
Hi @jasonmccallister,

What version of Ubuntu are you using in the instances?  If you are just using the default, there may be a change since default switch from 18.04 to 20.04 fairly recently (via cloud-images).

Also, I'm not what device is supposed to be on IRQ 3.  Could you please post the output of `sudo lspci -vvv` in the instance?

**Comment by jasonmccallister** [comment]
@townsend2010 I'm on `Ubuntu 20.04.1 LTS` and the content of `lspci` is:

```
00:00.0 Host bridge: Network Appliance Corporation Device 1275
	Control: I/O+ Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx-
	Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
	Latency: 0
	Expansion ROM at <unassigned> [disabled] [size=2K]
	Capabilities: [40] Express (v2) Root Port (Slot-), MSI 00
		DevCap:	MaxPayload 128 bytes, PhantFunc 0
			ExtTag- RBE-
		DevCtl:	CorrErr- NonFatalErr- FatalErr- UnsupReq-
			RlxdOrd- ExtTag- PhantFunc- AuxPwr- NoSnoop-
			MaxPayload 128 bytes, MaxReadReq 128 bytes
		DevSta:	CorrErr- NonFatalErr- FatalErr- UnsupReq- AuxPwr- TransPend-
		LnkCap:	Port #0, Speed 2.5GT/s, Width x1, ASPM L0s, Exit Latency L0s <64ns
			ClockPM- Surprise- LLActRep- BwNot- ASPMOptComp-
		LnkCtl:	ASPM Disabled; RCB 64 bytes Disabled- CommClk-
			ExtSynch- ClockPM- AutWidDis- BWInt- AutBWInt-
		LnkSta:	Speed 2.5GT/s (ok), Width x1 (ok)
			TrErr- Train- SlotClk- DLActive- BWMgmt- ABWMgmt-
		RootCap: CRSVisible-
		RootCtl: ErrCorrectable- ErrNon-Fatal- ErrFatal- PMEIntEna- CRSVisible-
		RootSta: PME ReqID 0000, PMEStatus- PMEPending-
		DevCap2: Completion Timeout: Not Supported, TimeoutDis-, NROPrPrP-, LTR-
			 10BitTagComp-, 10BitTagReq-, OBFF Not Supported, ExtFmt-, EETLPPrefix-
			 EmergencyPowerReduction Not Supported, EmergencyPowerReductionInit-
			 FRS-, LN System CLS Not Supported, TPHComp-, ExtTPHComp-, ARIFwd-
			 AtomicOpsCap: Routing- 32bit- 64bit- 128bitCAS-
		DevCtl2: Completion Timeout: 50us to 50ms, TimeoutDis-, LTR-, OBFF Disabled ARIFwd-
			 AtomicOpsCtl: ReqEn- EgressBlck-
		LnkCtl2: Target Link Speed: 2.5GT/s, EnterCompliance- SpeedDis-
			 Transmit Margin: Normal Operating Range, EnterModifiedCompliance- ComplianceSOS-
			 Compliance De-emphasis: -6dB
		LnkSta2: Current De-emphasis Level: -6dB, EqualizationComplete-, EqualizationPhase1-
			 EqualizationPhase2-, EqualizationPhase3-, LinkEqualizationRequest-

00:01.0 SCSI storage controller: Red Hat, Inc. Virtio block device
	Subsystem: Red Hat, Inc. Virtio block device
	Control: I/O+ Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx+
	Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
	Latency: 64
	Interrupt: pin A routed to IRQ 16
	Region 0: I/O ports at 2000 [size=128]
	Region 1: Memory at c0000000 (32-bit, non-prefetchable) [size=8K]
	Expansion ROM at c0002800 [virtual] [disabled] [size=2K]
	Capabilities: [40] MSI-X: Enable+ Count=2 Masked-
		Vector table: BAR=1 offset=00000000
		PBA: BAR=1 offset=00001000
	Capabilities: [4c] MSI: Enable- Count=1/1 Maskable- 64bit+
		Address: 0000000000000000  Data: 0000
	Kernel driver in use: virtio-pci

00:01.1 SATA controller: Intel Corporation 82801HR/HO/HH (ICH8R/DO/DH) 6 port SATA Controller [AHCI mode] (prog-if 01 [AHCI 1.0])
	Control: I/O+ Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx+
	Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
	Latency: 64
	Interrupt: pin B routed to IRQ 24
	Region 5: Memory at c0002000 (32-bit, non-prefetchable) [size=1K]
	Expansion ROM at c0003000 [virtual] [disabled] [size=2K]
	Capabilities: [40] MSI: Enable+ Count=1/1 Maskable- 64bit+
		Address: 00000000fee00000  Data: 4025
	Kernel driver in use: ahci
	Kernel modules: ahci

00:02.0 Ethernet controller: Red Hat, Inc. Virtio network device
	Subsystem: Red Hat, Inc. Virtio network device
	Control: I/O+ Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx+
	Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
	Latency: 64
	Interrupt: pin A routed to IRQ 17
	Region 0: I/O ports at 2080 [size=32]
	Region 1: Memory at c0004000 (32-bit, non-prefetchable) [size=8K]
	Expansion ROM at c0003800 [virtual] [disabled] [size=2K]
	Capabilities: [40] MSI-X: Enable+ Count=3 Masked-
		Vector table: BAR=1 offset=00000000
		PBA: BAR=1 offset=00001000
	Capabilities: [4c] MSI: Enable- Count=1/1 Maskable- 64bit+
		Address: 0000000000000000  Data: 0000
	Kernel driver in use: virtio-pci

00:05.0 Network and computing encryption device: Red Hat, Inc. Virtio RNG
	Subsystem: Red Hat, Inc. Virtio RNG
	Control: I/O+ Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx+
	Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
	Latency: 64
	Interrupt: pin A routed to IRQ 20
	Region 0: I/O ports at 20a0 [size=32]
	Region 1: Memory at c0006000 (32-bit, non-prefetchable) [size=8K]
	Expansion ROM at c0008000 [virtual] [disabled] [size=2K]
	Capabilities: [40] MSI-X: Enable+ Count=2 Masked-
		Vector table: BAR=1 offset=00000000
		PBA: BAR=1 offset=00001000
	Capabilities: [4c] MSI: Enable- Count=1/1 Maskable- 64bit+
		Address: 0000000000000000  Data: 0000
	Kernel driver in use: virtio-pci

00:1f.0 ISA bridge: Intel Corporation 82371SB PIIX3 ISA [Natoma/Triton II]
	Control: I/O+ Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx-
	Status: Cap- 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
	Latency: 0
	Expansion ROM at c0008800 [virtual] [disabled] [size=2K]
```

**Comment by townsend2010** [comment]
Thanks!  How about the output of `cat /proc/interrupts` as well?

**Comment by jasonmccallister** [comment]
```
           CPU0       CPU1       CPU2       CPU3
  4:          0          0       1696          0   IO-APIC   4-edge      ttyS0
  8:          0          0          0          0   IO-APIC   8-edge      rtc0
  9:          0          0          0          0   IO-APIC   9-fasteoi   acpi
 24:          0          0          0        177   PCI-MSI 18432-edge      ahci[0000:00:01.1]
 25:          0          0          0          0   PCI-MSI 16384-edge      virtio0-config
 26:          0          0          0      90303   PCI-MSI 16385-edge      virtio0-req.0
 27:          0          0          0          0   PCI-MSI 32768-edge      virtio1-config
 28:          0          0      17276          0   PCI-MSI 32769-edge      virtio1-input.0
 29:          0      82353          0        199   PCI-MSI 32770-edge      virtio1-output.0
 30:          0          0          0          0   PCI-MSI 81920-edge      virtio2-config
 31:          0          6         23          0   PCI-MSI 81921-edge      virtio2-input
NMI:          0          0          0          0   Non-maskable interrupts
LOC:    2827357    3221417    3260038    2232675   Local timer interrupts
SPU:          0          0          0          0   Spurious interrupts
PMI:          0          0          0          0   Performance monitoring interrupts
IWI:          4          0          0         11   IRQ work interrupts
RTR:          0          0          0          0   APIC ICR read retries
RES:     145625     134227     135505     132791   Rescheduling interrupts
CAL:      31639      32032      36893       9009   Function call interrupts
TLB:       8232       8188       8133       8324   TLB shootdowns
TRM:          0          0          0          0   Thermal event interrupts
THR:          0          0          0          0   Threshold APIC interrupts
DFR:          0          0          0          0   Deferred Error APIC interrupts
MCE:          0          0          0          0   Machine check exceptions
MCP:         42         42         42         42   Machine check polls
ERR:          0
MIS:          0
PIN:          0          0          0          0   Posted-interrupt notification event
NPI:          0          0          0          0   Nested posted-interrupt event
PIW:          0          0          0          0   Posted-interrupt wakeup event
```

**Comment by jasonmccallister** [comment]
That is the one I have seen most often. It does not seem to affect the running instance, just doing anything that requires reading output while it occurs makes it unreadable, as it happens repeatedly.

**Comment by townsend2010** [comment]
Ok, this is looking to be some kernel issue on 20.04 when using Hyperkit.  Unfortunately, because we have to boot the kernel itself, simply `apt update` on the kernel still won't use a newer kernel.

What output is being affected by this?  Any output, like this is showing up in the TTY?

**Comment by jasonmccallister** [comment]
Here is an example of what I see when this occurs, I am running `exec` to call `journalctl` and it randomly occurred.

![Screen Shot 2020-10-05 at 5 20 35 PM](https://user-images.githubusercontent.com/5354908/95133372-393ba680-072f-11eb-958c-0f8c5b86c8ab.png)

**Comment by townsend2010** [comment]
Hi @jasonmccallister,

I've been trying to do some additional digging and this is certainly an issue w/ the kernel and either something in Hyperkit or some additional software installed in the instance.  I looked through your Nitro custom cloud-init and there are certainly a host of additional packages installed, so it may be something there.

Also, it could be a red herring, but does this tend to occur when you running the `make api` step in Nitro?

Lastly, which kernel is reported in `uname -r` in the instance?

That said, I'm running out of ideas and I don't think there is anything in the Multipass realm that we can do to address this.

**Comment by jasonmccallister** [comment]
@townsend2010 this occurs when running `nitro ssh` which is a wrapper around `multipass shell`.

Here is the `uname -r` output:

```
5.4.0-48-generic
```

**Comment by sisyphosloughs** [comment]
Same here with `5.4.0-53-generic`. The error occures when I open the multipass shell by the `Open shell` command on macOS. 
Because this messages interrupts typing, I cannot use my multipass instance any more.

**Comment by yangl1996** [comment]
Same here with `5.4.0-54-generic` kernel. Exiting the shell and reattaching using `multipass shell instance-name`.

The outputs are `do_IRQ: 0.39 No irq handler for vector`. It started when I'm compiling something, but I'm not sure how to reproduce it. Happy to test more and try to reproduce.

**Comment by xuzhe0205** [comment]
> Same here with `5.4.0-54-generic` kernel. Exiting the shell and reattaching using `multipass shell instance-name`.
> 
> The outputs are `do_IRQ: 0.39 No irq handler for vector`. It started when I'm compiling something, but I'm not sure how to reproduce it. Happy to test more and try to reproduce.

Same here. But I control+C to "cancel" this message, and I am able to proceed my work in multipass instance. But still, not sure what's happening here

**Comment by RSully** [comment]
I am running into this issue. I have just started using Multipass for the first time - a fresh install of the current version on my Mac running 10.15.7. Within minutes of creating a brand new instance of 20.04 and simply running a couple commands (in my case simply apt update / upgrade) these messages began appearing.

**Comment by schlenki** [comment]
I have the same issue on OSX, always the same failure message randomly popping up as posted before. other than that, multipass works beautifully for me :)

**Comment by lmatusek** [comment]
I have the same issue, and it is quite new.  The message:

Message from syslogd@primary at Jan 13 07:15:58 ...
 kernel:[   97.526663] do_IRQ: 4.36 No irq handler for vector

Till now it never appeared. Rare, but disturbing on console. 

multipass  1.5.0+mac
multipassd 1.5.0+Mac
macOS Big Sur 11.1

**Comment by denisyilmaz** [comment]
Same issue here with Nitro and Multipass. Multipass deletes local bound vendor folder and after restart it outputs

nitro 1.1.1
multipass 1.5.0+mac
multipassd 1.5.0+Mac
macOS Big Sur 11.1


Message from syslogd@nitro-dev at Jan 13 17:38:48 ...
 kernel:[   99.704253] do_IRQ: 3.37 No irq handler for vector

After "composer install" within `nitro ssh` I'm able to restart, but after few hours the instance crashes (and deletes the vendor folder again).

**Comment by townsend2010** [comment]
Hey all,

It seems this issue is cropping up more and more.  I've yet to reproduce on my MacBook, so in order to try to figure out anything common oh what triggers this, I need more data.  Would would be helpful is to know:
1. Mac hardware specs such as what type of Mac, CPU sku, amount of RAM, disk type and size.
2. macOS version (most reports already have this, but it's still helpful if not posted).
3. How the instance was launched such as any `--cpu`, `--mem`, and `--disk` options given and also the version of Ubuntu launched.
4. Any mounts defined in the instance.
5. Any additional software/packages installed in the instance.
6. The type of activity occurring in the instance when this is seen.

That's all I can think of for now and thanks!

**Comment by lmatusek** [comment]
Here is my profile, attached.
[profile.txt](https://github.com/canonical/multipass/files/5810197/profile.txt)

**Comment by schlenki** [comment]
Hello,
my specs:
1. MacBook Pro 13" Retina Mid 2018, 2.3 GHz Quad-Core i5-8259U, 16 GB @ 2133 MHz RAM, 512 GB Apple SSD
2. I use macOS Catalina Version 10.15.7
3. I launched as follows (but it also happened w/ other disk or ram launchs in the past) multipass launch -c 4 -m 4G -n test ; creating a Ubuntu 20.04 LTS VM
4. No mounts in VM
5. just ran sudo apt update, nothing else on the VM
6. after sudo apt update, the failure happens; after installing Go 1.15.6 and Singularity 3.7.0 (plus dependencies), it stopped happening

**Comment by townsend2010** [comment]
Aha, I was able to reproduce!  Looking at `journalctl`, it looks to be an error in the `ata` driver in the instance's kernel.  I'll see if I research this some more and see what is going on.

**Comment by townsend2010** [comment]
Quick update...it's the virtual DVD drive in the instance that is misbehaving on the virtual SATA bus.  Since we are loading the kernel directly, I'm going to experiment with some kernel command line parameters such `noapic`, etc. to see if it helps without any detectable regressions.

**Comment by schlenki** [comment]
Thank you for your very much appreciated work in this! :)

**Comment by Saviq** [comment]
Hey all, @townsend2010 has a package for you to try:

macOS build available: [multipass-1.7.0-dev.513.pr343+g46704338.mac-Darwin.pkg](https://multipass-ci.s3.amazonaws.com/pr343/multipass-1.7.0-dev.513.pr343%2Bg46704338.mac-Darwin.pkg)

**Comment by schlenki** [comment]
Installing this pkg I get the following fault printed on my terminal, trying to do a multipass launch command:

launch failed: cannot connect to the multipass socket                           
Please ensure multipassd is running and '/var/run/multipass_socket' is accessible

multipass version shows:
multipass  1.7.0-dev.513.pr343+g46704338.mac

**Comment by townsend2010** [comment]
After some testing, it looks like adding the `noapic` kernel option works around this.  I have not detected any regressions in behavior.  Once our Mac ci is unblocked and builds a test package, I'll post an updated link here.  The previous test package posted here is broken for unrelated reasons.

**Comment by townsend2010** [comment]
Hi all,

Here is a test package that is working for me: [multipass-1.7.0-dev.549.pr343+g83f3483e.mac-Darwin.pkg](https://github.com/canonical/multipass-private/suites/1880016263/artifacts/36778648)

Please try it out and report back if it fixing this issue for you.  Thanks!

**Comment by Saviq** [comment]
Here's a link you'll be able to access:

https://multipass-ci.s3.amazonaws.com/pr343/multipass-1.7.0-dev.549.pr343%2Bg83f3483e.mac-Darwin.pkg

**Comment by Saviq** [comment]
Hi all, has anyone had the chance to try the above package if it fixes the issue for you?

**Comment by lmatusek** [comment]
Hi,

I am sorry I haven’t got time to make any check till now. Now I am trying….

Br,

Laszlo

**Comment by lmatusek** [comment]
Hey,

I have a new one. It does not work at all.


launch failed: The following errors occurred:
primary: timed out waiting for response
Saving session...completed.

[Folyamat befejeződött]


Name                    State             IPv4             Image
primary                 Unknown           --               Ubuntu 20.04 LTS

Br,

Laszlo

**Comment by townsend2010** [comment]
Hi @lmatusek,

Which test package did you try?  It looks like your error pertains the the first test package posted here and is broken.  If you didn't, please use the test package in https://github.com/canonical/multipass/issues/1778#issuecomment-766891272.  I will remove the first link as to avoid any confusion.

**Comment by spurin** [comment]
Hi @townsend2010

I've got the same issue reported, I installed the version in the link and after doing so, my instances would no longer start.  I've since reverted back to the latest version.

Happy to help with testing this further.

Thanks

**Comment by townsend2010** [comment]
There hasn't been any activity in this for a while and haven't seen any other complaints, so I'm guessing a Big Sur update is really what fixed the issue.  I'm closing this, but if I'm wrong, please reopen :slightly_smiling_face:

**Comment by skitsanos** [comment]
Having the same issue on OSX BigSur with multipass 1.6.2+mac. Instance with 4Gb RAM , 8Gb disk, and 2 CPUs

```
kernel:[  132.061868] do_IRQ: 0.40 No irq handler for vector
```

**Comment by nathanielks** [comment]
I am also encountering this issue:
```
Message from syslogd@primary at Jul 20 11:56:22 ...
 kernel:[  101.393201] do_IRQ: 4.36 No irq handler for vector
```

Attempting to stop the instance also failed:
```
Stopping primary -[2021-07-20T12:00:19.044] [error] [primary] process error occurred Crashed
```

I was running an `npm install` for a project. 

Version: multipassd 1.7.0+mac

**Comment by townsend2010** [comment]
Hmm, seems this is coming back again, so will reopen.  I will try to resurrect the test package I had with a potential fix as no one confirmed whether it fixed the issue at hand or not.

**Comment by klinebch** [comment]
I'm seeing the same behavior on multiple machines, with both MacOS 10.15 Catalina and 11.5.1 BigSur both running multipass 1.7.0+mac.  In my case it seems to be happening primarily when I'm doing operations on a mounted volume from the host's home.  It's locked up with the error when running maven builds, tar operations, and docker launches.

` kernel:[   54.710540] do_IRQ: 1.39 No irq handler for vector`

All terminal sessions then freeze after the next command and the VM must be killed.

**Comment by alexclewontin** [comment]
I'm having a variation on this problem as well, although mine is 

`kernel:[ 54.710540] do_IRQ: 4.35 No irq handler for vector`

My host machine is a 15-inch 2018 MacBook Pro running Big Sur 11.5.2, and I am experiencing it on 1.7.0+mac (although I was also experiencing it on a previous version, possibly 1.6.4?). The VM I am running has 12 CPUs (same as my host), 8 GB of memory (of 32GB total memory) and a 32GB disk, plus a mounted volume. Running Ubuntu 20.04, image hash 97bb9f79af52.

As additional context, it seems to happen in bursts. I'll suddenly get 5 or 6 messages, all a couple seconds apart, but other times I'll get none.

**Comment by tuduongquyet** [comment]
I'm also facing the same issue
```
➜  ~ multipass launch -n focal -c 2 -d 50G -m 4G
Launched: focal                                                                 
➜  ~ multipass shell focal                       
Welcome to Ubuntu 20.04.3 LTS (GNU/Linux 5.4.0-84-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

 System information disabled due to load higher than 2.0


1 update can be applied immediately.
To see these additional updates run: apt list --upgradable


To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

ubuntu@focal:~$ sudo apt update 
Get:1 http://security.ubuntu.com/ubuntu focal-security InRelease [114 kB]           
Hit:2 http://archive.ubuntu.com/ubuntu focal InRelease                              
Get:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease [114 kB]
Get:4 http://security.ubuntu.com/ubuntu focal-security/main amd64 Packages [861 kB]
Get:5 http://archive.ubuntu.com/ubuntu focal-backports InRelease [101 kB]
Get:6 http://archive.ubuntu.com/ubuntu focal/universe amd64 Packages [8628 kB]
Get:7 http://security.ubuntu.com/ubuntu focal-security/main Translation-en [166 kB]
Get:8 http://security.ubuntu.com/ubuntu focal-security/main amd64 c-n-f Metadata [8692 B]
Get:9 http://security.ubuntu.com/ubuntu focal-security/restricted amd64 Packages [405 kB]           
Get:10 http://security.ubuntu.com/ubuntu focal-security/restricted Translation-en [58.1 kB]   
Get:11 http://security.ubuntu.com/ubuntu focal-security/restricted amd64 c-n-f Metadata [508 B]
Get:12 http://security.ubuntu.com/ubuntu focal-security/universe amd64 Packages [639 kB]  
Get:13 http://security.ubuntu.com/ubuntu focal-security/universe Translation-en [101 kB]        
Get:14 http://security.ubuntu.com/ubuntu focal-security/universe amd64 c-n-f Metadata [12.4 kB]
Get:15 http://security.ubuntu.com/ubuntu focal-security/multiverse amd64 Packages [21.9 kB]  
Get:16 http://security.ubuntu.com/ubuntu focal-security/multiverse Translation-en [4948 B]
Get:17 http://security.ubuntu.com/ubuntu focal-security/multiverse amd64 c-n-f Metadata [540 B]
Get:18 http://archive.ubuntu.com/ubuntu focal/universe Translation-en [5124 kB]            
Get:19 http://archive.ubuntu.com/ubuntu focal/universe amd64 c-n-f Metadata [265 kB]                                                                                                                       
Get:20 http://archive.ubuntu.com/ubuntu focal/multiverse amd64 Packages [144 kB]                                                                                                                           
Get:21 http://archive.ubuntu.com/ubuntu focal/multiverse Translation-en [104 kB]                                                                                                                           
Get:22 http://archive.ubuntu.com/ubuntu focal/multiverse amd64 c-n-f Metadata [9136 B]                                                                                                                     
Get:23 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 Packages [1205 kB]                                                                                                                        
Get:24 http://archive.ubuntu.com/ubuntu focal-updates/main Translation-en [258 kB]                                                                                                                         
Get:25 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 c-n-f Metadata [14.2 kB]                                                                                                                  
Get:26 http://archive.ubuntu.com/ubuntu focal-updates/restricted amd64 Packages [442 kB]                                                                                                                   
Get:27 http://archive.ubuntu.com/ubuntu focal-updates/restricted amd64 c-n-f Metadata [504 B]                                                                                                              
Get:28 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 Packages [855 kB]                                                                                                                     
Get:29 http://archive.ubuntu.com/ubuntu focal-updates/universe Translation-en [181 kB]                                                                                                                     
Get:30 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 c-n-f Metadata [18.8 kB]                                                                                                              
Get:31 http://archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 Packages [24.6 kB]                                                                                                                  
Get:32 http://archive.ubuntu.com/ubuntu focal-updates/multiverse Translation-en [6776 B]                                                                                                                   
Get:33 http://archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 c-n-f Metadata [620 B]                                                                                                              
Get:34 http://archive.ubuntu.com/ubuntu focal-backports/main amd64 Packages [2568 B]                                                                                                                       
Get:35 http://archive.ubuntu.com/ubuntu focal-backports/main Translation-en [1120 B]                                                                                                                       
Get:36 http://archive.ubuntu.com/ubuntu focal-backports/main amd64 c-n-f Metadata [400 B]                                                                                                                  
Get:37 http://archive.ubuntu.com/ubuntu focal-backports/restricted amd64 c-n-f Metadata [116 B]                                                                                                            
Get:38 http://archive.ubuntu.com/ubuntu focal-backports/universe amd64 Packages [5800 B]                                                                                                                   
Get:39 http://archive.ubuntu.com/ubuntu focal-backports/universe Translation-en [2068 B]                                                                                                                   
Get:40 http://archive.ubuntu.com/ubuntu focal-backports/universe amd64 c-n-f Metadata [276 B]                                                                                                              
Get:41 http://archive.ubuntu.com/ubuntu focal-backports/multiverse amd64 c-n-f Metadata [116 B]                                                                                                            
Fetched 19.9 MB in 8s (2437 kB/s)                                                                                                                                                                          
Reading package lists... 84%
Message from syslogd@focal at Sep 10 14:13:25 ...
 kernel:[   66.329410] do_IRQ: 1.39 No irq handler for vector

Message from syslogd@focal at Sep 10 14:13:25 ...
 kernel:[   66.330263] do_IRQ: 1.39 No irq handler for vector
Reading package lists... Done
Building dependency tree... 50%
Message from syslogd@focal at Sep 10 14:13:26 ...
 kernel:[   66.646569] do_IRQ: 1.39 No irq handler for vector
Building dependency tree       
Reading state information... Done
2 packages can be upgraded. Run 'apt list --upgradable' to see them.
ubuntu@focal:~$ 
Message from syslogd@focal at Sep 10 14:13:31 ...
 kernel:[   71.659784] do_IRQ: 1.39 No irq handler for vector

Message from syslogd@focal at Sep 10 14:13:31 ...
 kernel:[   71.662901] do_IRQ: 1.39 No irq handler for vector

Message from syslogd@focal at Sep 10 14:13:31 ...
 kernel:[   72.051115] do_IRQ: 1.39 No irq handler for vector

Message from syslogd@focal at Sep 10 14:13:41 ...
 kernel:[   82.184523] do_IRQ: 1.39 No irq handler for vector

Message from syslogd@focal at Sep 10 14:13:41 ...
 kernel:[   82.187472] do_IRQ: 1.39 No irq handler for vector

Message from syslogd@focal at Sep 10 14:13:41 ...
 kernel:[   82.508082] do_IRQ: 1.39 No irq handler for vector

Message from syslogd@focal at Sep 10 14:14:12 ...
 kernel:[  113.457540] do_IRQ: 1.39 No irq handler for vector

Message from syslogd@focal at Sep 10 14:14:12 ...
 kernel:[  113.460700] do_IRQ: 1.39 No irq handler for vector
```

**Comment by mhucka** [comment]
I'm new to multipass, but am facing this same issue on an iMac running macOS 10.15 (Catalina) with Ubuntu 20.04.4 LTS.

```
Message from syslogd@mike at May 13 18:19:01 ...
 kernel:[  400.956619] do_IRQ: 0.40 No irq handler for vector
```

**Comment by townsend2010** [comment]
Hey @mhucka,

You can switch to the `qemu` driver which is much better supported than Hyperkit.  To do this, do `$ multipass set local.driver=qemu`.  Please note that any existing instances will be hidden, but not lost and can be seen again by switching back to the `hyperkit` driver.

**Comment by townsend2010** [comment]
Anyone running into this issue who is running macOS 10.15 or greater can switch to the `qemu` driver as outlined in https://github.com/canonical/multipass/issues/1778#issuecomment-1126604492.

**Comment by mhucka** [comment]
@townsend2010 thanks for the frighteningly fast reply. I've tried it by doing
```
multipass set local.driver=qemu
```
before recreating everything, but now it seems to be stuck. I used `-vvvv` with `multipass launch` and here is where it's been for 15+ minutes now (far longer than it's ever taken for this step) – where "mike" is the name of the instance:

```
Launching mike
[2022-05-13T18:39:40.924] [trace] [url downloader] Found https://cdimage.ubuntu.com/ubuntu-core/16/stable/current/SHA256SUMS in cache: true
[2022-05-13T18:39:41.013] [trace] [url downloader] Found https://cdimage.ubuntu.com/ubuntu-core/18/stable/current/SHA256SUMS in cache: true
[2022-05-13T18:39:41.014] [trace] [url downloader] Found https://cloud-images.ubuntu.com/buildd/releases/bionic/release/SHA256SUMS in cache: true
[2022-05-13T18:39:41.015] [trace] [url downloader] Found https://cloud-images.ubuntu.com/buildd/releases/focal/release/SHA256SUMS in cache: true
[2022-05-13T18:39:41.016] [trace] [url downloader] Found https://cloud-images.ubuntu.com/buildd/daily/jammy/current/SHA256SUMS in cache: true
[2022-05-13T18:39:41.018] [trace] [url downloader] Found https://cloud-images.ubuntu.com/releases/streams/v1/index.json in cache: true
[2022-05-13T18:39:41.048] [trace] [url downloader] Found https://cloud-images.ubuntu.com/releases/streams/v1/com.ubuntu.cloud:released:download.json in cache: true
[2022-05-13T18:39:41.129] [trace] [url downloader] Found https://cdimage.ubuntu.com/ubuntu-core/appliances/streams/v1/index.json in cache: true
[2022-05-13T18:39:41.130] [trace] [url downloader] Found https://cdimage.ubuntu.com/ubuntu-core/appliances/streams/v1/com.ubuntu.appliances:released:download.json in cache: true
[2022-05-13T18:39:41.137] [debug] [qemu-img] [22450] started: qemu-img info /var/root/Library/Caches/multipassd/vault/images/focal-20220505/ubuntu-20.04-server-cloudimg-amd64.img
Configuring mike /
```

Update: I killed it via ^C and now cannot do anything with multipass, because it says it cannot connect to the multipass socket:
```
list failed: cannot connect to the multipass socket
Please ensure multipassd is running and '/var/run/multipass_socket' is accessible
```

And I cannot figure out how to restart it ...

**Comment by townsend2010** [comment]
Since this occurs with the Hyperkit driver and it has been deprecated in 1.12 and will be removed in 1.13, I'm going to close this issue as "won't fix."
